lib/internal/util.js

```diff
@@ -60,6 +60,9 @@ exports._deprecate = function(fn, msg) {
   var warned = false;
   function deprecated() {
     warned = exports.printDeprecationMessage(msg, warned, deprecated);
+    if (new.target) {
+      return new (Function.prototype.bind.apply(fn, arguments));
```

Comment:
I think it's important to make the deprecated `class` correctly subclassable. Can you change this line to `return Reflect.construct(fn, arguments, new.target)` and add a test that verifies that ?